src/libsync/propagatecommonzsync.cpp

```diff
@@ -0,0 +1,320 @@
+Q_LOGGING_CATEGORY(lcZsyncPut, "sync.networkjob.zsync.put", QtInfoMsg)
+
+bool isZsyncPropagationEnabled(OwncloudPropagator *propagator, SyncFileItemPtr item)
+{
+    ConfigFile cfg;
+    if (!propagator->account()->capabilities().zsyncSupported()) {
+        qCInfo(lcPropagator) << "[zsync disabled] Lack of server support.";
+        item->_isZsync = false;
+        return false;
+    }
+    if (item->_remotePerm.hasPermission(RemotePermissions::IsMounted) ||
+        item->_remotePerm.hasPermission(RemotePermissions::IsMountedSub)) {
+        qCInfo(lcPropagator) << "[zsync disabled] External storage not supported.";
+        item->_isZsync = false;
+        return false;
+    }
+    if (!(cfg.deltaSyncEnabled() ||
+          propagator->_deltaSyncEnabled.fetchAndAddAcquire(0))) {
+        qCInfo(lcPropagator) << "[zsync disabled] Client configuration option.";
+        item->_isZsync = false;
+        return false;
+    }
+    if (item->_size < cfg.deltaSyncMinFileSize() ||
+        item->_size < propagator->_deltaSyncMinFileSize.fetchAndAddAcquire(0)) {
+        qCInfo(lcPropagator) << "[zsync disabled] File size is smaller than minimum.";
+        item->_isZsync = false;
+        return false;
+    }
+
+    item->_isZsync = true;
+    return true;
+}
+
+void ZsyncSeedJob::seed()
+{
+    // Create a temporary file to use with zsync_begin()
+    QTemporaryFile zsyncControlFile;
+    zsyncControlFile.open();
+    zsyncControlFile.write(_zsyncData.constData(), _zsyncData.size());
+    zsyncControlFile.flush();
+
+    int fileHandle = zsyncControlFile.handle();
+    FILE* f = fdopen(dup(fileHandle), "r");
+    setvbuf(f, NULL, _IONBF, 0);
+    rewind(f);
+
+    struct zsync_state *zs;
+    if ((zs = zsync_begin(f)) == NULL) {
+        QString errorString = tr("Unable to parse zsync file.");
+        emit failedSignal(errorString);
+        return;
+    }
+
+    zsyncControlFile.close();
+    fclose(f);
+
+    std::string tmp = _zsyncFilePath.toStdString();
+    const char * fname = tmp.c_str();
+    {
+        /* Simple uncompressed file - open it */
+        FILE *f = fopen(fname, "r");
+        if (!f) {
+            QString errorString = tr("Unable to open file.");
+            emit failedSignal(errorString);
+            return;
+        }
+        else {
+
+            /* Give the contents to libzsync to read, to find any content that
+             * is part of the target file. */
+            qCInfo(lcZsyncSeed) << "Reading seed file:" << _zsyncFilePath;
+            zsync_submit_source_file(zs, f, false, _type == ZsyncMode::download ? false : true);
+
+            /* And close */
+            if (fclose(f) != 0) {
+                QString errorString = tr("Unable to close file.");
+                emit failedSignal(errorString);
+                return;
+            }
+        }
+    }
+
+    emit finishedSignal(zs);
+}
+
+/* stream_error(function, stream) - Exit with IO-related error message */
```

Comment:
doc -> header, also for the other functions

Reply:
Copy/pasted from zsync as-is. Fixed.